Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -49,11 +58,23 @@ extension MongoCollection {
         let rp = options?.readPreference?._readPreference
         let pipeline: Document = ["pipeline": pipeline]
 
-        guard let cursor = mongoc_collection_aggregate(
-            self._collection, MONGOC_QUERY_NONE, pipeline._bson, opts?._bson, rp) else {
-            fatalError("Couldn't get cursor from the server")
+        let conn = try self._client.connectionPool.checkOut()
+        let cursor: OpaquePointer = self.withMongocCollection(from: conn) { collPtr in
+            guard let cursor = mongoc_collection_aggregate(collPtr,
+                                                           MONGOC_QUERY_NONE,
+                                                           pipeline._bson,
+                                                           opts?._bson,
+                                                           rp) else {
+                fatalError("Couldn't get cursor from the server")
```

Comment:
starting to feel like this message should be a constant

Reply:
added a constant `failedToRetrieveCursorMessage`